mem/file.go - Fix some races in accessing fields of FileData

corentone opened the pull request:
* Splitting SetModeTime to avoid double locking
* Adding locks all over the place.

Potential fix for #133

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spf13/afero?pullRequest=134) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/spf13/afero?pullRequest=134) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spf13/afero?pullRequest=134) it.</sub>

bep commented on mem/file.go:
```diff
@@ -74,14 +74,24 @@ func CreateDir(name string) *FileData {
 
 func SetMode(f *FileData, mode os.FileMode) {
+	f.Lock()
+	defer f.Unlock()
```
No need to use `defer` when the method is so short.

bep replied:
Also, it would be great if you added a test that demonstrates this problem.

corentone commented:
@bep thanks for the review and sorry for the delay addressing them.
Let me know if you have any comment, I also added some UT for the funcs I changed which are kind of boringly verbose...

mbertschler commented:
I verified that the new tests fail with data races without this patch.

mbertschler commented:
Fixes #133
